classes/village/VillageManager.php

```diff
@@ -1498,6 +1528,49 @@ public static function enactProposal(System $system, User $player, int $proposal
                     SET `region_locations`.`occupying_village_id` = NULL
                     WHERE `regions`.`village` = {$proposal['village_id']} AND `region_locations`.`occupying_village_id` = {$proposal['target_village_id']}");
                 break;
+            case self::PROPOSAL_TYPE_OFFER_TRADE:
+                // check at war
+                if ($player->village->isEnemy($proposal['target_village_id'])) {
```

Comment:
Hmm, should we explicitly make this allies only?

Reply:
Yeah I'm not sure why I wanted the ability to trade with neutral.